src/catch2/reporters/catch_reporter_json.cpp

```diff
@@ -0,0 +1,222 @@
+
+//              Copyright Catch2 Authors
+// Distributed under the Boost Software License, Version 1.0.
+//   (See accompanying file LICENSE.txt or copy at
+//        https://www.boost.org/LICENSE_1_0.txt)
+
+// SPDX-License-Identifier: BSL-1.0
+//
+#include <catch2/catch_test_case_info.hpp>
+#include <catch2/catch_test_spec.hpp>
+#include <catch2/catch_version.hpp>
+#include <catch2/interfaces/catch_interfaces_config.hpp>
+#include <catch2/internal/catch_string_manip.hpp>
+#include <catch2/reporters/catch_reporter_json.hpp>
+
+namespace Catch {
+    JsonReporter::JsonReporter( ReporterConfig&& config ):
+        StreamingReporterBase{ CATCH_MOVE( config ) } {
+        m_objectWriters.emplace( m_stream );
+        m_writers.emplace( Writer::Object );
+    }
+
+    JsonReporter::~JsonReporter() {
+        while ( !m_writers.empty() ) {
+            switch ( m_writers.top() ) {
+            case Writer::Object:
+                endObject();
+                break;
+            case Writer::Array:
+                endArray();
+                break;
+            }
+        }
+    }
+
+    void JsonReporter::startArray() {
+        if ( !isInside( Writer::Array ) ) { return; }
+        m_arrayWriters.emplace( m_arrayWriters.top().writeArray() );
+        m_writers.emplace( Writer::Array );
+    }
+    void JsonReporter::startArray( std::string const& key ) {
+        if ( !isInside( Writer::Object ) ) { return; }
+        m_arrayWriters.emplace(
+            m_objectWriters.top().write( key ).writeArray() );
+        m_writers.emplace( Writer::Array );
+    }
+
+    void JsonReporter::startObject() {
+        if ( !isInside( Writer::Array ) ) { return; }
+        m_objectWriters.emplace( m_arrayWriters.top().writeObject() );
+        m_writers.emplace( Writer::Object );
+    }
+    void JsonReporter::startObject( std::string const& key ) {
+        if ( !isInside( Writer::Object ) ) { return; }
+        m_objectWriters.emplace(
+            m_objectWriters.top().write( key ).writeObject() );
+        m_writers.emplace( Writer::Object );
+    }
+
+    void JsonReporter::endObject() {
+        if ( !isInside( Writer::Object ) ) { return; }
+        m_objectWriters.pop();
+        m_writers.pop();
+    }
+    void JsonReporter::endArray() {
+        if ( !isInside( Writer::Array ) ) { return; }
+        m_arrayWriters.pop();
+        m_writers.pop();
+    }
+
+    bool JsonReporter::isInside( Writer writer ) {
+        return not m_writers.empty() && m_writers.top() == writer;
```

Comment:
```suggestion
        return !m_writers.empty() && m_writers.top() == writer;
```

MSVC doesn't like the keywords without including `<ciso646>`.